src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_7001_8000.cpp

```diff
@@ -4832,5 +4832,21 @@ GUI_TEST_CLASS_DEFINITION(test_7896) {
     GTUtilsMsaEditor::checkMsaEditorWindowIsActive();
 }
 
+GUI_TEST_CLASS_DEFINITION(test_7907) {
+    GTFileDialog::openFile(testDir + "_common_data/regression/7907/example.fa");
+    GTUtilsSequenceView::checkSequenceViewWindowIsActive();
+
+    FindEnzymesDialogFillerSettings settings;
+    settings.enzymes = QStringList{ "AbaSI" };
+    GTUtilsDialog::waitForDialog(new FindEnzymesDialogFiller(settings));
+    GTWidget::click(GTWidget::findWidget("Find restriction sites_widget"));
+    GTUtilsTaskTreeView::waitTaskFinished();
+
+    GTGlobals::FindOptions options;
+    options.failIfNotFound = false;
+    auto a = GTUtilsAnnotationsTreeView::findItem("AbaSI", nullptr, options);
```

Comment:
This should be an XML test (it checks the algorithm only)

Could you please also add a test when an enzyme Ns cross a circular sequence junction point (I this the current solution is not safe for circulars)